### PR TITLE
fix(iamRole): iamPermissionsBoundary and inlinePolicies bug fixes

### PR DESCRIPTION
**Pull request by hjaraujof**
## Changes/solution

- Fixes permission boundary bug
- Added null checks

### CODE REVIEW

File: src/services/iamRole/data.ts [hunk 1]
```diff
@@ -224,14 +228,13 @@ export const listIamRoles = async ({
...
                   Tags: tags.find(t => t?.RoleName === RoleName)?.Tags || {},
-                  PermissionsBoundaryArn:
-                    PermissionsBoundary.PermissionsBoundaryArn,
-                  InlinePolicies: roleAuthorizationDetails
-                    .find(rAD => rAD.RoleName === RoleName)
-                    .RolePolicyList.map(rPl => ({
-                      name: rPl.PolicyName,
-                      document: rPl.PolicyDocument,
-                    })),
+                  ...(PermissionsBoundary?.PermissionsBoundaryArn
```
Comment (tyler-dunkel) [inline src/services/iamRole/data.ts:231]: why an object here if we are only storing one property?

Comment (hjaraujof) [reply src/services/iamRole/data.ts:231]: This small check deconstructs into the property we need to add if the value is not undefined. This way, I don't introduce an empty/undefined property to the result

File: src/services/iamRole/data.ts [hunk 2]
```diff
@@ -224,14 +228,13 @@ export const listIamRoles = async ({
...
+                    ? {
+                        PermissionsBoundaryArn:
```
Comment (tyler-dunkel) [inline src/services/iamRole/data.ts:233]: should this not be `permissionsBoundaryArn` not `PermissionsBoundaryArn`?

Comment (tyler-dunkel) [reply src/services/iamRole/data.ts:233]: ahh I see this is the data file not the format file. ignore

**Review by tyler-dunkel — APPROVED** [review]

**Pull request merged**